src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,50 @@
+using Distances
+
+export statistical_complexity
+
+
+"""
+    statistical_complexity(x, est, distance = JSDivergence(), normalize = true, entropy_func=Renyi())
+
+This function calculates the statistical complexity of a time series `x` as defined by Rosso et al. [^Rosso2007].
+It calculates the statistical complexity based on permutation patterns. The statistical complexity ``C_{JS}[P]`` 
+is defined based on the distribution ``P`` of ordinal patterns
+```math
+C_{JS}[P] = Q_J[P, P_e] H_S[P]
+```
+where ``Q_J`` is originally the normalized Jensen-Shannon divergence of the ordinal distribution to a uniform distribution,
+and ``H_S`` is the normalized permutation entropy.
+
+In this implementation, the distance measure defaults to the Jensen-Shannon divergence, but other (semi-)metrics can also be
+chosen. The type of entropy defaults to the Renyi entropy.
+
+If `normalize = true`, the complexity is divided by the maximum distance of the ordinal distribution to a uniform distribution,
+otherwise this normalization step is skipped.
+
+[^Rosso2007] Rosso et al. (2007). Distinguishing noise from chaos. https://doi.org/10.1103/PhysRevLett.99.154102
+"""
+function statistical_complexity(x::AbstractArray, est::ProbabilitiesEstimator;
+                                distance::SemiMetric = JSDivergence(),
+                                normalize::Bool = true,
+                                entropy_func::Entropy = Renyi()) 
+
+    # get probabilities from data
+    probs = probabilities(x, est)
+    # fill with zeros for empty bins, we need this to compare to a uniform distribution of m! patterns
+    probs = cat(probs, zeros(factorial(est.m)-length(probs)), dims=1)
+    # generate uniform distribution with N = length(p) bins
+    uniform = ones(length(probs)) * 1 / length(probs)
```

Comment:
This is ok, but I'd rather prefer something like

```julia
N = length(probs)
p_uniform = Probabilities(repeat([1 / N], N))
```

which is easier to read and removes the need for a comment. Wrapping with `Probabilities` immediately makes it clear that this is a distribution. Since `Probabilities` is just a wrapper around `Vector`, this should "just work" when evaluating distances later in the too (given that all necessary methods for `Probabilities` have been implemented).